src/hooks/useGeolocationHandler.ts

```diff
@@ -0,0 +1,82 @@
+
+import { Matcher, SelectableStaticFilter, useSearchActions, useSearchState } from '@yext/search-headless-react';
+import { executeSearch } from '../utils/search-operations';
+import { getUserLocation } from '../utils/location-operations';
+import { useCallback } from 'react';
+
+const LOCATION_FIELD_ID = 'builtin.location';
+const METERS_PER_MILE = 1609.344;
+
+/**
+ * The props for {@link useGeolocationHandler} hook.
+ *
+ * @internal
+ */
+interface GeolocationhandlerProps {
```

Comment:
Could we leave `Props` out of the name here? When I see `Props`, I expect a React Component, not a hook.

Reply:
sure, I can update it to `GeolocationhandlerArgs` instead